Doc/whatsnew/3.13.rst

```diff
@@ -265,223 +264,238 @@ Improved Error Messages
 
-* The cycle garbage collector is now incremental.
-  This means that maximum pause times are reduced
-  by an order of magnitude or more for larger heaps.
+Free-threaded execution allows for full utilization of the available
+processing power by running threads in parallel on available CPU cores.
+While not all software will benefit from this automatically, programs
+designed with threading in mind will run faster on multi-core hardware.
+**The free-threaded mode is experimental** and work is ongoing to improve it:
+expect some bugs and a substantial single-threaded performance hit.
+Free-threaded builds of CPython support optionally running with the GIL
+enabled at runtime using the environment variable :envvar:`PYTHON_GIL` or
+the command-line option :option:`-X gil`.
 
-Support For Mobile Platforms
-----------------------------
+To check if the current interpreter supports free-threading, :option:`python -VV <-V>`
+and :attr:`sys.version` contain "experimental free-threading build".
+The new :func:`!sys._is_gil_enabled` function can be used to check whether
+the GIL is actually disabled in the running process.
 
-* iOS is now a :pep:`11` supported platform. ``arm64-apple-ios``
-  (iPhone and iPad devices released after 2013) and
-  ``arm64-apple-ios-simulator`` (Xcode iOS simulator running on Apple Silicon
-  hardware) are now tier 3 platforms.
+C-API extension modules need to be built specifically for the free-threaded
+build. Extensions that support running with the :term:`GIL` disabled should
+use the :c:data:`Py_mod_gil` slot. Extensions using single-phase init should
+use :c:func:`PyUnstable_Module_SetGIL` to indicate whether they support
+running with the GIL disabled. Importing C extensions that don't use these
+mechanisms will cause the GIL to be enabled, unless the GIL was explicitly
+disabled with the :envvar:`PYTHON_GIL` environment variable or the
+:option:`-X gil=0` option.
+pip 24.1 or newer is required to install packages with C extensions in the
+free-threaded build.
 
-  ``x86_64-apple-ios-simulator`` (Xcode iOS simulator running on older x86_64
-  hardware) is not a tier 3 supported platform, but will be supported on a
-  best-effort basis.
+.. seealso::
 
-  See :pep:`730`: for more details.
+   :pep:`703` "Making the Global Interpreter Lock Optional in CPython"
+   contains rationale and information surrounding this work.
 
-  (PEP written and implementation contributed by Russell Keith-Magee in
-  :gh:`114099`.)
 
 .. _whatsnew313-jit-compiler:
 
-Experimental JIT Compiler
--------------------------
+A just-in-time compiler for CPython
+-----------------------------------
+
+When CPython is configured and built using
+the :option:`!--enable-experimental-jit` option,
+a just-in-time (JIT) compiler is added which may speed up some Python programs.
+On Windows, use ``PCbuild/build.bat --experimental-jit`` to enable the JIT
+or ``--experimental-jit-interpreter`` to enable the Tier 2 interpreter.
```

Comment:
Can we mention here ``Tools/jit/README.md `` which provides more information about the "external tools" that a user needs to build CPython with JIT?

Reply:
Personally, I'd rather not link to a GitHub document as a primary source of documentation -- we should have at least a basic level of content regarding the JIT somewhere in the docs (perhaps ``Doc/using/configure.rst`` or a new ``Doc/using/jit.rst``) that we can point to in *What's New*.

However, the JIT is special in that it is experimental, so perhaps that link would be appropriate. I'll wait for @brandtbucher to comment in general as he might have a view here.

A